File: api/src/main/java/club/yunzhi/log/entity/Course.java
package club.yunzhi.log.entity;

import javax.persistence.*;

/**
 * 课程
 */
@Entity
public class Course {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", nullable = false)
    private Long id;

    /**
     * 名称
     */
    private String name;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getCourseCredit() {
        return courseCredit;
    }

    public void setCourseCredit(int courseCredit) {
        this.courseCredit = courseCredit;
    }

    public int getExperimentalCredit() {
        return experimentalCredit;
    }

    public void setExperimentalCredit(int experimentalCredit) {
        this.experimentalCredit = experimentalCredit;
    }

    public String getExaminationMethod() {
        return examinationMethod;
    }

    public void setExaminationMethod(String examinationMethod) {
        this.examinationMethod = examinationMethod;
    }

    /**
     * 学期
     */
    @ManyToOne
    @JoinColumn(name = "semester_id")
    private Semester semester;

    /**
     * 课程学分
     */
    private int courseCredit;

    /**
     * 实验学分
     */
    private int experimentalCredit;

    /**
     * 考核方式
     */
    private String examinationMethod;

    public Semester getSemester() {
        return semester;
    }

    public void setSemester(Semester semester) {
        this.semester = semester;
    }
}

File: web/src/entity/course.ts
/**
 * 课程
 */
import {Semester} from './semester';
import {Klass} from './Klass';
import {Teacher} from './teacher';
import {Major} from './major';

export class Course {
  /**
   * id
   */
  id?: number;
  /**
   * 名称
   */
  name?: string;


  /**
   * 学期
   */
  semester: Semester;

  /**
   * 课程学分
   */
  courseCredit: number;

  /**
   * 实验学分
   */
  experimentalCredit: number;

  /**
   * 考核方式
   */
  examinationMethod: string;

  /**
   * 教师
   */
  teacher?: Teacher;

  /**
   * 开课专业
   */
  majors?: Major[];

  constructor(data = {} as {
    id?: number,
    name?: string,
    semester?: Semester,
    courseCredit?: number,
    experimentalCredit?: number,
    examinationMethod?: string,
    teacher?: Teacher,
    majors?: Major[]
  }) {
    this.id = data.id;
    this.name = data.name;
    this.semester = data.semester;
    this.courseCredit = data.courseCredit;
    this.experimentalCredit = data.experimentalCredit;
    this.examinationMethod = data.examinationMethod;
    this.teacher = data.teacher;
    this.majors = data.majors;
  }

}
